File: Data Structures & Algorithms/maximum-subarray/submission-2.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = INT_MIN;
        for (int i = 0; i < nums.size(); i++) {
            int currSum = 0;
            for (int j = i; j < nums.size(); j++) {
                currSum += nums[j];
                ans = max(ans, currSum);
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        self.ans = 0

        def solve(node, level):
            if not node:
                return
            self.ans = max(self.ans, level)
            solve(node.left, level+1)
            solve(node.right, level+1)
        solve(root, 1)
        return self.ans